#!/bin/bash
#SBATCH -J "job_name"
#SBATCH -o log_%j
#SBATCH -c 1
#SBATCH -p medium
#SBATCH --mail-type=FAIL
#SBATCH --mail-user=YOUREMAIL
#SBATCH --time=06-00:00
#SBATCH --mem=08G
# Move to directory where job was submitted
cd $SLURM_SUBMIT_DIR
#########################################################
#AUTHOR: Q. Rougemont
#Last update: 28-05-2019
#Purpose: Script To select SNP variant from gatk-4.1.2.0
#INPUT:   1 vcf file obtained after running GenotypeGVCFs
#INPUT:   1 fasta file (reference genome)
#OUTPUT : 1 vcf file with SNPs only (for all individuals) 
########################################################

#Global variables
gvcfall=$1        #input whole genome vcf file
if [ -z $gvcfall ];
then
    echo "error no input vcf provided"
    echo "input file expected to be generated by joint genotying (GenotypeGVCFs)"
    exit
fi
#
#ceate folder:
OUTFOLDER="12-snp_VCF"
if [ ! -d "$OUTFOLDER" ]
then 
    echo "creating out-dir"
    mkdir "$OUTFOLDER"
fi

FILE_PATH=$(pwd)

##################Â run gatk ########################################
echo "############# Running GATK ###########"
echo "# extracting SNPs from whole vcf.gz #"
gatk --java-options "-Xmx10G" \
    SelectVariants \
    -V "$gvcfall" \
    --select-type-to-include SNP \
    -O "$FILE_PATH"/"$OUTFOLDER"/GVCFall_SNPs.vcf.gz 
